atlasdb-config/src/main/java/com/palantir/atlasdb/factory/Leaders.java

```diff
@@ -111,20 +113,23 @@ public static LocalPaxosServices createLocalServices(
         Map<PingableLeader, HostAndPort> otherLeaders = generatePingables(
                 remotePaxosServerSpec.remoteLeaderUris(), sslSocketFactory, userAgent);
 
-        ExecutorService executor = Executors.newCachedThreadPool(new ThreadFactoryBuilder()
-                .setNameFormat("atlas-leaders-%d")
-                .setDaemon(true)
-                .build());
+        InstrumentedExecutorService instrumentedExecutor = new InstrumentedExecutorService(
+                Executors.newCachedThreadPool(new ThreadFactoryBuilder()
+                        .setNameFormat("atlas-leaders-%d")
+                        .setDaemon(true)
+                        .build()),
+                AtlasDbMetrics.getMetricRegistry());
 
-        PaxosProposer proposer = createPaxosProposer(ourLearner, acceptors, learners, config.quorumSize(), executor);
+        PaxosProposer proposer = createPaxosProposer(ourLearner, acceptors, learners, config.quorumSize(),
+                instrumentedExecutor);
```

Comment:
I'm thinking something along lines of the following, though ideally some of these could be fixed sized pools and apply back pressure appropriately rather than thread spawning to death.

```
        PaxosProposer proposer = createPaxosProposer(ourLearner, acceptors, learners, config.quorumSize(),
                new InstrumentedExecutorService(
                        Executors.newCachedThreadPool(new ThreadFactoryBuilder()
                                .setNameFormat("atlas-proposer-%d")
                                .setDaemon(true)
                                .build()),
                        AtlasDbMetrics.getMetricRegistry(),
                        MetricRegistry.name(PaxosProposer.class, "executor")));

        ExecutorService leadersExecutor = new InstrumentedExecutorService(
                Executors.newCachedThreadPool(new ThreadFactoryBuilder()
                        .setNameFormat("atlas-leaders-%d")
                        .setDaemon(true)
                        .build()),
                AtlasDbMetrics.getMetricRegistry(),
                MetricRegistry.name(PaxosLeaderElectionService.class, "executor"));
```